### PR TITLE
Supports the new API auth scheme which switched away from JWTs

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/2123

This needs to land in sync with https://github.com/mozilla/addons-server/pull/4960

### CODE REVIEW

File: src/core/server/base.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { createApiError } from 'core/api';
 import ServerHtml from 'core/containers/ServerHtml';
 import { prefixMiddleWare, trailingSlashesMiddleware } from 'core/middleware';
 import { convertBoolean } from 'core/utils';
-import { setClientApp, setLang, setJwt, setUserAgent } from 'core/actions';
+import { setClientApp, setLang, setAuthToken, setUserAgent } from 'core/actions';
```
Comment (tofumatt) [inline src/core/server/base.js]: Nitpick but we tend to alphabetise imports. But like, only fix if you need to fix something else too.

File: tests/client/core/api/test_api.js [hunk 2]
```diff
@@ -446,7 +446,7 @@ describe('api', () => {
   });
 
   describe('login', () => {
-    const response = { token: 'use.this.jwt' };
+    const response = { token: 'use:this:auth-token' };
```
Comment (tofumatt) [inline tests/client/core/api/test_api.js]: This token is different from the others, why is that?

File: tests/client/core/reducers/test_api.js [hunk 3]
```diff
@@ -11,15 +11,16 @@ describe('api reducer', () => {
...
+    assert.deepEqual(
+      api({ foo: 'bar' }, actions.setAuthToken(token)),
+      { foo: 'bar', token });
   });
 
-  it('clears the JWT on log out', () => {
+  it('clears the auth token on log out', () => {
     assert.deepEqual(
-      api({ lang: 'fr', clientApp: 'firefox', token: 'secret' },
+      api({ lang: 'fr', clientApp: 'firefox', token: 'user:auth:token' },
```
Comment (tofumatt) [inline tests/client/core/reducers/test_api.js]: Dunno if using `signedInApiState` here makes more sense, but I guess it doesn't buy us anything. Just consistency, really.

Comment (kumar303) [inline tests/client/core/reducers/test_api.js]: Not sure how I missed that, thanks

**Pull request merged**